triples-generator/src/main/scala/ch/datascience/triplesgenerator/reprovisioning/TriplesRemover.scala

```diff
@@ -29,47 +32,43 @@ private trait TriplesRemover[Interpretation[_]] {
   def removeAllTriples(): Interpretation[Unit]
 }
 
-private object TriplesRemover {
-  val TriplesRemovalBatchSize: Long = 50000
-}
-
 private class IOTriplesRemover(
+    removalBatchSize:        Long Refined Positive,
     rdfStoreConfig:          RdfStoreConfig,
     logger:                  Logger[IO],
     timeRecorder:            SparqlQueryTimeRecorder[IO]
 )(implicit executionContext: ExecutionContext, contextShift: ContextShift[IO], timer: Timer[IO])
     extends IORdfStoreClient(rdfStoreConfig, logger, timeRecorder)
     with TriplesRemover[IO] {
 
-  import TriplesRemover._
-  import cats.implicits._
   import eu.timepit.refined.auto._
   import io.circe.Decoder
 
-  import scala.util.Try
-
   override def removeAllTriples(): IO[Unit] =
-    queryExpecting[Long](findTriplesCount)(countDecoder) flatMap {
-      case 0 => IO.unit
-      case _ =>
+    queryExpecting(checkIfEmpty)(storeEmptyFlagDecoder) flatMap { isEmpty =>
+      if (isEmpty) IO.unit
+      else
         for {
           _ <- updateWitNoResult(removeTriplesBatch)
```

Comment:
missing an 'h' in updateWit**h**NoResult

Reply:
Done